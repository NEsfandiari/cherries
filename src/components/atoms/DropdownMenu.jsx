import React, { Component } from 'react'
import styled from 'styled-components'
import { NavLink } from '../atoms'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  .dropdown {
    margin: 0;
  }
  .dropdown-content {
    display: none;
    position: absolute;
    background-color: #fbe5e9;
    border-radius: 2px;
    box-shadow: 1px 1px 10px 0 rgba(46, 61, 73, 0.2);
    margin-top: 2rem;
    padding: 0.5rem;
    animation-duration: 0.5s;
    z-index: 3;
    a {
      padding: 0.2rem;
      font-size: 0.8rem;
    }
    .links {
      display: flex;
      flex-direction: column;
    }
    animation: fadein 0.25s;
    @keyframes fadein {
      from {
        opacity: 0;
      }
      to {
        opacity: 1;
      }
    }
  }
`

class DropdownMenu extends Component {
  state = {
    display: false,
  }
  showMenu = () => {
    this.setState({ display: true })
  }
  hideMenu = () => {
    this.setState({ display: false })
  }

  render() {
    const display = this.state.display ? 'initial' : 'none'
    const links = this.props.links.map((link, i) => {
      if (link.route) {
        return (
          <NavLink to={link.route} key={link.name}>
            {link.name}
          </NavLink>
        )
      } else {
        return (
          <NavLink to="/" onClick={this.props.logOutUser} key={link.name}>
            {link.name}
          </NavLink>
        )
      }
    })
    return (
      <Container onMouseLeave={this.hideMenu}>
        <NavLink className="dropdown" onMouseEnter={this.showMenu} to="">
          <img
            style={{
              margin: 5,
              maxWidth: 28,
              maxHeight: 28,
            }}
            src={this.props.icon}
            alt="nav-bar-icon"
          />
        </NavLink>
        <div className="dropdown-content" style={{ display: display }}>
          <div className="links">{links}</div>
        </div>
      </Container>
    )
  }
}

export default DropdownMenu
