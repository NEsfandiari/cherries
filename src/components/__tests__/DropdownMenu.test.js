import React from 'react'
import renderer from 'react-test-renderer'
import DropdownMenu from '../atoms/DropdownMenu'

describe('DropdownMenu', () =>
  it('renders correctly', () => {
    const icon =
      '//images.ctfassets.net/4yejk6rho2rp/6u7gsl9D3Oo46iYwW04y8c/66b1a6434576a65e23d5501ee03fb1a8/help-icon.png?w=800&q=50'

    const links = [
      {
        name: 'FAQ',
        route: 'faq',
      },
      {
        name: 'Help',
        route: 'faq',
      },
      {
        name: 'Returns',
        route: 'faq',
      },
      {
        name: 'team@lipslut.com',
        route: 'contact',
      },
    ]
    const tree = renderer
      .create(<DropdownMenu icon={icon} links={links} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))
