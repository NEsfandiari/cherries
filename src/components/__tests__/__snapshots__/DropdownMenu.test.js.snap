// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu renders correctly 1`] = `
<div
  className="sc-VigVT fuTuEz"
  onMouseLeave={[Function]}
>
  <a
    className="dropdown sc-htpNat eMXSJb"
    href=""
    onMouseEnter={[Function]}
  >
    <img
      alt="nav-bar-icon"
      src="//images.ctfassets.net/4yejk6rho2rp/6u7gsl9D3Oo46iYwW04y8c/66b1a6434576a65e23d5501ee03fb1a8/help-icon.png?w=800&q=50"
      style={
        Object {
          "margin": 5,
          "maxHeight": 28,
          "maxWidth": 28,
        }
      }
    />
  </a>
  <div
    className="dropdown-content"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="links"
    >
      <a
        className="sc-htpNat eMXSJb"
        href="faq"
      >
        FAQ
      </a>
      <a
        className="sc-htpNat eMXSJb"
        href="faq"
      >
        Help
      </a>
      <a
        className="sc-htpNat eMXSJb"
        href="faq"
      >
        Returns
      </a>
      <a
        className="sc-htpNat eMXSJb"
        href="contact"
      >
        team@lipslut.com
      </a>
    </div>
  </div>
</div>
`;
